### DISCUSSION TITLE
[Migration] Auto-migration DB error when upgrade langfuse version

### DISCUSSION THREAD

**Discussion by charlieviettq** [langfuse/langfuse#2353] [Support] [answered]
Please help me with this issue:

When I upgrade Langfuse from version v2.44.0 to v2.48.1, I encounter problems with the auto-migration of the database using Prisma:
```
Prisma schema loaded from packages/shared/prisma/schema.prisma
Datasource "db": PostgreSQL database "cake_langfuse", schema "public" at "127.0.0.1:5432"

201 migrations found in prisma/migrations

Error: P3009

Migrate found failed migrations in the target database; new migrations will not be applied. Read more about how to resolve migration issues in a production database: https://pris.ly/d/migrate-resolve
The `20240604133339_score_data_type_add_boolean` migration started at 2024-06-06 11:10:36.489938 UTC failed

Applying database migrations failed. This is mostly caused by the database being unavailable.
Exiting...
```
Please help me to provide some context to fix this smoothly. All auto-migrations before were successful, but this case has failed.

**Comment by marcklingen** [comment]
Which Postgres version do you run?

Can you try upgrading to https://github.com/langfuse/langfuse/releases/tag/v2.54.0? This version fixed some migration issues in previous versions which were related to not running on the public schema or running Postgres 12

**Reply by charlieviettq** [reply]
I run on postgrest version 12

**Reply by marcklingen** [reply]
Do you still have this issue when trying to upgrade to the latest version?

**Reply by charlieviettq** [reply]
I just released to version 2.54.0, but still have issue

**Reply by charlieviettq** [reply]
I will try to upgrade to latest version right now, but may be its not effect because rls notes is not relate to

**Reply by marcklingen** [reply]
can you share issue log here? especially the log column of the migrations postgres table

**Reply by charlieviettq** [reply]
<img width="1409" alt="image" src="https://github.com/langfuse/langfuse/assets/82230029/5413ea96-f6de-4c44-a126-11a097e89c64">

**Reply by charlieviettq** [reply]
here bros

**Reply by charlieviettq** [reply]
is this useful?

**Reply by marcklingen** [reply]
did you get any server logs that explain why this migration failed?

**Reply by charlieviettq** [reply]
hmm, no bros, only raise this log: 
```Prisma schema loaded from packages/shared/prisma/schema.prisma
Datasource "db": PostgreSQL database "cake_langfuse", schema "public" at "127.0.0.1:5432"

201 migrations found in prisma/migrations

Error: P3009

Migrate found failed migrations in the target database; new migrations will not be applied. Read more about how to resolve migration issues in a production database: https://pris.ly/d/migrate-resolve
The `20240604133339_score_data_type_add_boolean` migration started at 2024-06-06 11:10:36.489938 UTC failed

Applying database migrations failed. This is mostly caused by the database being unavailable.
Exiting...
```


Prisma error code P3009: "migrate found failed migrations in the target database, new migrations will not be applied. Read more about how to resolve migration issues in a production database: https://pris.ly/d/migrate-resolve
{details}"

**Reply by marcklingen** [reply, marked as answer]
This is the content of the migration

```
ALTER TYPE "ScoreDataType" ADD VALUE 'BOOLEAN';
```

Is `BOOLEAN` part of `ScoreDataType` in your db? If yes, mark the migration as finished in the migrations table (add date in finished_at, and applied_step_count=1). If no, delete the row from the migrations table.

**Reply by charlieviettq** [reply]
Thank you so much. I deleted the row from the migrations table and It's worked with me!

**Reply by marcklingen** [reply]
perfect, thanks for confirming! still do not know what went wrong here during the migration but glad you were able to resolve it this way.
